src/generate/peripheral.rs

```diff
@@ -163,13 +163,15 @@ pub fn render(
         return Ok(out);
     }
 
+    let defaults = p.defaults.derive_from(defaults);
```

Comment:
I'd rather use `register_properties` or `default_register_properties` here to express the meaning of this "default". The same for the struct fields. This property set is called `registerPropertiesGroup` in SVD spec.